protocol/app/app.go

```diff
@@ -956,6 +981,7 @@ func New(
 		upgrade.NewAppModule(app.UpgradeKeeper),
 		evidence.NewAppModule(app.EvidenceKeeper),
 		ibc.NewAppModule(app.IBCKeeper),
+		ica.NewAppModule(nil, &app.ICAHostKeeper),
```

Comment:
why do we need pass `ICAHostKeeper` by address here? seems like an anti-pattern based on the other `NewAppModule(...)` here

Reply:
The `ica.NewAppModule()` [specifically takes in pointers](https://github.com/cosmos/ibc-go/blob/cea0a82b49a0e18d3999ae81825fdbeea2fc6b5c/modules/apps/27-interchain-accounts/module.go#L117).

Some of these Cosmos SDK modules take in copy, some take in pointers. Haven't had time to dig into whether this is intended pattern or just inconsistency